crates/cargo-util-schemas/src/manifest/mod.rs

```diff
@@ -1503,14 +1503,29 @@ impl TomlLint {
             Self::Config(config) => config.priority,
         }
     }
+
+    pub fn sub_config(&self) -> Option<&TomlLintSubConfig> {
+        match self {
+            Self::Level(_) => None,
+            Self::Config(config) => config.sub_config.as_ref(),
+        }
+    }
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 #[serde(rename_all = "kebab-case")]
 pub struct TomlLintConfig {
-    pub level: TomlLintLevel,
+    pub level: Option<TomlLintLevel>,
```

Comment:
imo this should be at least a separate commit, probably a separate PR

Reply:
It's already a separate commit, albeit dependent on the first one, and the reason it's dependant on the first one is to force the `level` if no `check-cfg` is provided, which can only be done when we have knowledge of `check-cfg`.

Reply:
That is a detail too buried in everything which makes me feel more strongly this needs to be its own PR to make sure we know what semantics we are agreeing to.

Reply:
Will do that in a follow-up PR.